jtlibrary/python/jtmodules/src/jtmodules/segment_secondary_cp.py

```diff
@@ -0,0 +1,338 @@
+       thresholding method. lambda is a regularization parameter; see the help for
+       the setting for more details. Propagation of secondary object labels is
+       by the shortest path to an adjacent primary object from the starting
+       ("seeding") primary object. The seed-to-pixel distances are calculated
+       as the sum of absolute differences in a 3x3 (8-connected) image
+       neighborhood, combined with lambda via sqrt(differences^2 +
+       lambda^2).
+    -  watershed_gradient: This method uses the watershed algorithm
+       (Vincent and Soille, 1991) to assign pixels to the primary objects
+       which act as seeds for the watershed. In this variant, the watershed
+       algorithm operates on the Sobel transformed image which computes an
+       intensity gradient. This method works best when the image intensity
+       drops off or increases rapidly near the boundary between cells.
+    -  watershed_image: This method is similar to the above, but it
+       uses the inverted intensity of the image for the watershed. The areas
+       of lowest intensity will be detected as the boundaries between cells.
+       This method works best when there is a saddle of relatively low
+       intensity at the cell-cell boundary.
+    -  Distance: In this method, the edges of the primary objects are
+       expanded a specified distance to create the secondary objects. For
+       example, if nuclei are labeled but there is no stain to help locate
+       cell edges, the nuclei can simply be expanded in order to estimate
+       the cell's location. This is often called the "doughnut" or "annulus"
+       or "ring" approach for identifying the cytoplasm. There are two
+       methods that can be used:
+       -  distance_n: In this method, the image of the secondary
+          staining is not used at all; the expanded objects are the final
+          secondary objects.
+       -  distance_b: Thresholding of the secondary staining image
+          is used to eliminate background regions from the secondary
+          objects. This allows the extent of the secondary objects to be
+          limited to a certain distance away from the edge of the primary
+          objects without including regions of background.
+    References
+    ^^^^^^^^^^
+    Jones TR, Carpenter AE, Golland P (2005) "Voronoi-Based Segmentation of
+    Cells on Image Manifolds", ICCV Workshop on Computer Vision for
+    Biomedical Image Applications, 535-543.
+    Vincent L, Soille P (1991) "Watersheds in Digital Spaces: An Efficient
+    Algorithm Based on Immersion Simulations", IEEE Transactions on Pattern
+    Analysis and Machine Intelligence, Vol. 13, No. 6, 583-598
+    '''
+
+    if not np.any(primary_label_image == 0):
+        secondary_label_image = primary_label_image
+    else:
+        n_objects = len(np.unique(primary_label_image[1:]))
+        logger.info(
+            'primary label image has %d objects',
+            n_objects - 1
```

Comment:
```suggestion
            'primary label image has {} objects'.format(n_objects - 1)
```
% is being deprecated.
Is there a reason why we reducing the number of objects by 1?

Reply:
Yes, 0 is not an object